bigchaindb/__init__.py

```diff
@@ -5,24 +5,51 @@
 # PORT_NUMBER = reduce(lambda x, y: x * y, map(ord, 'BigchainDB')) % 2**16
 # basically, the port number is 9984
 
-_database_rethinkdb = {
-    'backend': os.environ.get('BIGCHAINDB_DATABASE_BACKEND', 'rethinkdb'),
+
+_base_database_rethinkdb = {
     'host': os.environ.get('BIGCHAINDB_DATABASE_HOST', 'localhost'),
     'port': int(os.environ.get('BIGCHAINDB_DATABASE_PORT', 28015)),
     'name': os.environ.get('BIGCHAINDB_DATABASE_NAME', 'bigchain'),
-    'connection_timeout': 5000,
-    'max_tries': 3,
 }
 
-_database_mongodb = {
-    'backend': os.environ.get('BIGCHAINDB_DATABASE_BACKEND', 'mongodb'),
+# The following variable is used by `bigchaindb configure` to
+# prompt the user for database values. We cannot rely on
+# _base_database_rethinkdb.keys() or _base_database_mongodb.keys()
+# because dicts are unordered. I tried to configure
+
+_base_database_rethinkdb_keys = ('host', 'port', 'name')
+
+_base_database_mongodb = {
     'host': os.environ.get('BIGCHAINDB_DATABASE_HOST', 'localhost'),
     'port': int(os.environ.get('BIGCHAINDB_DATABASE_PORT', 27017)),
     'name': os.environ.get('BIGCHAINDB_DATABASE_NAME', 'bigchain'),
     'replicaset': os.environ.get('BIGCHAINDB_DATABASE_REPLICASET', 'bigchain-rs'),
+    'ssl': bool(os.environ.get('BIGCHAINDB_DATABASE_SSL', False)),
+    'login': os.environ.get('BIGCHAINDB_DATABASE_LOGIN'),
+    'password': os.environ.get('BIGCHAINDB_DATABASE_PASSWORD')
+}
+
+_base_database_mongodb_keys = ('host', 'port', 'name', 'replicaset',
+                               'ssl', 'login', 'password')
```

Comment:
> Oh sorry, I overlooked the comment.

No problem :dancer: 

I was thinking the same, but building one is quite awful because you have to pass a list of "pairs" unfortunately :sob:

```python
# from the docs

>>> # regular unsorted dictionary
>>> d = {'banana': 3, 'apple': 4, 'pear': 1, 'orange': 2}

>>> # dictionary sorted by key
>>> OrderedDict(sorted(d.items(), key=lambda t: t[0]))
OrderedDict([('apple', 4), ('banana', 3), ('orange', 2), ('pear', 1)])

>>> # dictionary sorted by value
>>> OrderedDict(sorted(d.items(), key=lambda t: t[1]))
OrderedDict([('pear', 1), ('orange', 2), ('banana', 3), ('apple', 4)])

>>> # dictionary sorted by length of the key string
>>> OrderedDict(sorted(d.items(), key=lambda t: len(t[0])))
OrderedDict([('pear', 1), ('apple', 4), ('orange', 2), ('banana', 3)])
```

For us it would be:
```python
OrderedDict([
    ('backend', os.environ.get('BIGCHAINDB_DATABASE_BACKEND', 'rethinkdb')),
    ('host', os.environ.get('BIGCHAINDB_DATABASE_HOST', 'localhost')),
    ('port', int(os.environ.get('BIGCHAINDB_DATABASE_PORT', 28015))),
    ('name', os.environ.get('BIGCHAINDB_DATABASE_NAME', 'bigchain'))
])
```

It might be fine tho, but I really don't like the ﻿ａｅｓｔｈｅｔｉｃ.